### ISSUE TITLE
[BUG] crc setup fails during Windows installation even after multiple reboots

### ISSUE THREAD

**Issue by brucedkyle** [crc-org/crc#3539]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V 
  * Did you run `crc setup` before starting it (Yes) But it fails
  * Running CRC on: Laptop

## CRC version
```bash
CRC version: 2.14.0+868d96cd
OpenShift version: 4.12.1
Podman version: 4.3.1
```
  
## CRC status
```bash
Machine does not exist. Use 'crc start' to create it
```

## CRC config
```bash
- consent-telemetry                     : yes
- memory                                : 16384
- pull-secret-file                      : C:\Users\myusername\Documents\afoldername\pull-secret.txt
```

## Host Operating System
```bash
Host Name:                
OS Name:                   Microsoft Windows 11 Enterprise
OS Version:                10.0.22000 N/A Build 22000
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          N/A
Registered Organization:   N/A
Product ID:                00330-80000-00000-AA131
Original Install Date:     7/12/2022, 11:52:47 AM
System Boot Time:          3/9/2023, 2:32:57 PM
System Manufacturer:       LENOVO
System Model:              20SUS04P00
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 165 Stepping 2 GenuineIntel ~2592 Mhz
BIOS Version:              LENOVO N30ET49W (1.32 ), 12/14/2022
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume1
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-08:00) Pacific Time (US & Canada)
Total Physical Memory:     32,519 MB
Available Physical Memory: 21,596 MB
Virtual Memory: Max Size:  39,687 MB
Virtual Memory: Available: 26,143 MB
Virtual Memory: In Use:    13,544 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\LAPTOP-SLIV8CB4
Hotfix(s):                 5 Hotfix(s) Installed.
                           [01]: KB5022505
                           [02]: KB5007575
                           [03]: KB5012170
                           [04]: KB5022836
                           [05]: KB5022369
Network Card(s):           5 NIC(s) Installed.
                           [01]: Intel(R) Wi-Fi 6 AX201 160MHz
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.86.1
                                 IP address(es)
                                 [01]: 192.168.86.47
                                 [02]: fe80::587a:c286:a015:a90f
                           [02]: Intel(R) Ethernet Connection (11) I219-V
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [03]: Cisco AnyConnect Virtual Miniport Adapter for Windows x64
                                 Connection Name: Ethernet 2
                                 Status:          Hardware not present
                           [04]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (crc)
                                 Status:          Media disconnected
                           [05]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

Have installed crc locally on Windows by running "C:\Users\myusername\Downloads\crc-windows-installer\crc-windows-amd64.msi" .

when I run `crc setup`, I get 

```text
INFO Using bundle path C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Adding logon user to crc-users and Hyper-V admins group
INFO Will run as admin: adding current user to crc-users and Hyper-V admins group
Please reboot your system and run 'crc setup' to complete the setup process
```

I reboot.

I then run `crc setup` and get the same response. Crc setup does not create C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle

`crc start` responds

```text
file 'C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle' does not exist
```
The desktop in the tooltray is loaded but will not start either.

What have I missed?

### Expected

`crc setup` should have created
`C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle`

### Actual

```text
file 'C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle' does not exist
```

### Logs

Before gather the logs try following if that fix your issue
```bash
$ crc delete -f
Machine does not exist. Use 'crc start' to create it
$ crc cleanup
INFO Removing the daemon task
INFO Removing hosts file records added by CRC
INFO Removing pull secret from the keyring
INFO Removing older logs
INFO Removing CRC Machine Instance directory
INFO Removing crc's virtual machine
Cleanup finished
$ crc setup
INFO Using bundle path C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Adding logon user to crc-users and Hyper-V admins group
INFO Will run as admin: adding current user to crc-users and Hyper-V admins group
Please reboot your system and run 'crc setup' to complete the setup process
$ crc start --log-level debug
DEBU CRC version: 2.14.0+868d96cd
DEBU OpenShift version: 4.12.1
DEBU Podman version: 4.3.1
DEBU Running 'crc start'
DEBU Total memory of system is 34098995200 bytes
DEBU Got bundle path: C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId).ReleaseId'
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
file 'C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle' does not exist
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

```
file 'C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle' does not exist
```

**Comment by anjannath** [comment]
> I then run crc setup and get the same response.

@brucedkyle did you run `crc setup` again after the reboot? 
the bundle (`C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle`) will be downloaded by `crc setup` its around 3GB so takes some time depending on your connection.

**Comment by brucedkyle** [comment]
Thank you for your kind response.

Yes. 

**I reboot.** 

I then get the same response from `crc setup`

```text
PS C:\> crc setup
INFO Using bundle path C:\Users\myusername\.crc\cache\crc_hyperv_4.12.1_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Adding logon user to crc-users and Hyper-V admins group
INFO Will run as admin: adding current user to crc-users and Hyper-V admins group
**Please reboot your system and run 'crc setup' to complete the setup process**
```

The tool tray is there, but will not start. There is no crc in **Hyper-v Manager**

Thoughts?

**Comment by albfan** [comment]
My understanding is this is related to some problem with hyper-v, groups, users and permissions

```
PS C:\Users\Alberto> crc setup --log-level debug
DEBU CRC version: 2.15.0+72256c3c
DEBU OpenShift version: 4.12.5
DEBU Podman version: 4.3.1
DEBU Running 'crc setup'
INFO Using bundle path C:\Users\Alberto\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
DEBU Running 'Get-LocalGroupMember -Group 'crc-users' -Member 'Alberto''
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: Get-LocalGroupMember : No se encontr� la entidad de seguridad Alberto.
En l�nea: 1 Car�cter: 43
+ ... yContinue'; Get-LocalGroupMember -Group 'crc-users' -Member 'Alberto' ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Alberto:String) [Get-LocalGroupMember], PrincipalNotFoundException
    + FullyQualifiedErrorId : PrincipalNotFound,Microsoft.PowerShell.Commands.GetLocalGroupMemberCommand

DEBU exit status 1
INFO Adding logon user to crc-users and Hyper-V admins group
INFO Will run as admin: adding current user to crc-users and Hyper-V admins group
DEBU Running 'C:\Users\Alberto\AppData\Local\Temp\crcScripts3066326615\runAsAdmin.ps1'
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId).ReleaseId'
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
Please reboot your system and run 'crc setup' to complete the setup process
PS C:\Users\Alberto> crc start -p 'C:\Users\Alberto\Downloads\pull-secret.txt'
file 'C:\Users\Alberto\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle' does not exist
PS C:\Users\Alberto><
```

**Comment by brucedkyle** [comment]
This answer was incredibly helpful. I am not sure how to fix it in the code. But I was able to reproduce the same error as Alberto.

```text
DEBU Running 'Get-LocalGroupMember -Group 'crc-users' -Member '6J1943897''
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: Get-LocalGroupMember : Principal 6J1943897 was not found.
At line:1 char:43
+ ... yContinue'; Get-LocalGroupMember -Group 'crc-users' -Member '6J194389 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (6J1943897:String) [Get-LocalGroupMember], PrincipalNotFoundException
    + FullyQualifiedErrorId : PrincipalNotFound,Microsoft.PowerShell.Commands.GetLocalGroupMemberCommand

DEBU exit status 1
```

When I run `Get-LocalGroupMember -Group 'crc-users' -Member '6J1943897'` by itself in PowerShell, this call will recreate the error. 

```
Get-LocalGroupMember : Principal 6J1943897 was not found.
At line:1 char:1
+ Get-LocalGroupMember -Group 'crc-users' -Member '6J1943897'
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (6J1943897:String) [Get-LocalGroupMember], PrincipalNotFoundException
    + FullyQualifiedErrorId : PrincipalNotFound,Microsoft.PowerShell.Commands.GetLocalGroupMemberCommand
```

The fix is to use the domain name along with the username to create the Member. The following call works:

```powershell
 Get-LocalGroupMember -Group 'crc-users' -Member 'GMX\6J1943897'
```
responds with:

```
ObjectClass Name          PrincipalSource
----------- ----          ---------------
User        GMX\6J1943897 AzureAD
```

**It sure looks like that in my case, you need both `$Env:UserDomain` and ` $Env:UserName` to build the `-Member`  parameter.**

It looks like `-Member` is being assembled in `/pkg/crc/preflight/preflight_checks_windows.go` lines 103 and 110.

`_, _, err := powershell.Execute(fmt.Sprintf("Get-LocalGroupMember -Group 'crc-users' -Member '%s'", username()))`

I am running... 
```
CRC version: 2.14.0+868d96cd
OpenShift version: 4.12.1
Podman version: 4.3.1
```

**Is `func username()` on line 157 in **preflight_windows.go** returning the username in a way that my computer is not expecting?**

Are we onto something? Suggestions?

**Comment by anjannath** [comment]
@brucedkyle with `2.15.0` we had another patch to fix this issue (#3523), it'd very helpful if you could test that version and see if that fixes the issue.

from @albfan logs it seems, reboot was not performed.

The correct flow is: install MSI -> run `crc setup` -> reboot -> run `crc setup` and then `crc start` should work

**Comment by albfan** [comment]
@anjannath I did several times, @brucedkyle is pointing right solution

```
crc version
CRC version: 2.15.0+72256c3c
OpenShift version: 4.12.5
Podman version: 4.3.1
```

still:

```
> Get-LocalGroupMember -Group 'crc-users' -Member 'Alberto'
Get-LocalGroupMember : No se encontró la entidad de seguridad Alberto.
En línea: 1 Carácter: 1
+ Get-LocalGroupMember -Group 'crc-users' -Member 'Alberto'
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Alberto:String) [Get-LocalGroupMember], PrincipalNotFoundException
    + FullyQualifiedErrorId : PrincipalNotFound,Microsoft.PowerShell.Commands.GetLocalGroupMemberCommand
```

but 

```
> Get-LocalGroupMember -Group 'crc-users'

ObjectClass Name                    PrincipalSource
----------- ----                    ---------------
Usuario     LENOVO-THINKPAD\Alberto Local
```

will update the crc and see if now it works

**Comment by cfergeau** [comment]
> will update the crc and see if now it works

Be aware of https://github.com/crc-org/crc/issues/3543 though, it's being resolved, but no ETA on this.

**Comment by albfan** [comment]
I tried to fake this set with:

```
> $env:UserName="$env:UserDomain\$env:UserName"
PS C:\Users\Alberto\projects> echo $env:UserName
LENOVO-THINKPAD\Alberto
```

But still it doesn't allow to continue

```
PS C:\Users\Alberto\projects> crc setup
INFO Using bundle path C:\Users\Alberto\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if C:\Users\Alberto\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle exists
INFO Getting bundle for the CRC executable
INFO Downloading bundle: C:\Users\Alberto\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle...
checksum mismatch
```

@cfergeau thanks for info, will wait for that then

**Comment by brucedkyle** [comment]
@cfergeau, @anjannath and @albfan many thanks for your quick and kind attention.

Unfortunately I tried the latest version and still go the same responses.

`crc version`

```
CRC version: 2.15.0+72256c3c
OpenShift version: 4.12.5
Podman version: 4.3.1
```

Is that the version you needed me to check?

`crc setup` after a reboot

```
INFO Using bundle path C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Adding logon user to crc-users and Hyper-V admins group
INFO Will run as admin: adding current user to crc-users and Hyper-V admins group
Please reboot your system and run 'crc setup' to complete the setup process
```
`crc start --log-level debug ` replies

```
DEBU CRC version: 2.15.0+72256c3c
DEBU OpenShift version: 4.12.5
DEBU Podman version: 4.3.1
DEBU Running 'crc start'
DEBU Total memory of system is 34098995200 bytes
DEBU Got bundle path: C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId).ReleaseId'
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
```

In the meantime, I'm dusting off my dev skills and starting up some basic GoLang. It is really cool. And what you are doing unifying scripts, PowerShell, and the OS is awesome. 

I should try it on my personal laptop too to see if it reproduces. There is probably something different about using my corporate laptop. My current workaround is to try to install crc in a RHEL VM in Hyper-V (with nested virtual machines, which I think might work).

Of course, your suggestions are welcome too. And I appreciate the support.

**Comment by cfergeau** [comment]
Could you run again `crc setup --log-level debug` with 2.15.0? And is `Running '(Get-ItemProperty...')` the very last line printed when you run `crc start`?

**Comment by brucedkyle** [comment]
Thanks for your quick reply, @cfergeau

`crc setup --log-level debug` returns

```text
DEBU CRC version: 2.15.0+72256c3c
DEBU OpenShift version: 4.12.5
DEBU Podman version: 4.3.1
DEBU Running 'crc setup'
INFO Using bundle path C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
DEBU Running 'Get-LocalGroupMember -Group 'crc-users' -Member '6J1943897''
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: Get-LocalGroupMember : Principal 6J1943897 was not found.
At line:1 char:43
+ ... yContinue'; Get-LocalGroupMember -Group 'crc-users' -Member '6J194389 ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (6J1943897:String) [Get-LocalGroupMember], PrincipalNotFoundException
    + FullyQualifiedErrorId : PrincipalNotFound,Microsoft.PowerShell.Commands.GetLocalGroupMemberCommand

DEBU exit status 1
INFO Adding logon user to crc-users and Hyper-V admins group
INFO Will run as admin: adding current user to crc-users and Hyper-V admins group
DEBU Running 'C:\Users\6J1943~1\AppData\Local\Temp\crcScripts4282647081\runAsAdmin.ps1'
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId).ReleaseId'
DEBU Running '(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID'
Please reboot your system and run 'crc setup' to complete the setup process
```

Next, `Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion" -Name ReleaseId` returns

```text
ReleaseId    : 2009
PSPath       : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion
PSParentPath : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT
PSChildName  : CurrentVersion
PSDrive      : HKLM
PSProvider   : Microsoft.PowerShell.Core\Registry
```

Next, `(Get-ItemProperty -Path "HKLM:\SOFTWARE\Microsoft\Windows NT\CurrentVersion").EditionID` returns

```text
Enterprise
```

My apologies, I should have included the last line of `crc start --log-level debug` again for clarity.

```text
file 'C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle' does not exist
```

**Comment by anjannath** [comment]
@brucedkyle Hi created #3559 , i see the issue is actually checking if the username is part of the group, the msi is successfully able to add the user to the required group but its the pre-flight check that fails. in the PR i try to check if the user is part of the group with a string comparison and not using the `-Member` flag for the powershell cmdlet.

Could you please uninstall your existing CRC and give the [artifact](https://github.com/crc-org/crc/suites/11783396050/artifacts/614639158) from that PR a try and see if that fixes the issue, thanks!

**Comment by brucedkyle** [comment]
Nice work figuring out the issue. @anjannath

I deleted the previous version and started up the .msi that you mentioned.

Unfortunately, the installer returned `Failed to add user to group (-2147023509 6J1943897 crc-users)`. The status bar was just beginning to show some green.

Sounds like we've figured out something about the issue. Is it that weird username that my company laptop is using perhaps? Baffled.

**Comment by anjannath** [comment]
> Unfortunately, the installer returned Failed to add user to group (-2147023509 6J1943897 crc-users). The status bar was just beginning to show some green.

i think this is because i removed the domain component while adding the group in msi, https://github.com/anjannath/crc/commit/e3d0a3ca0707e5df58d12ddfc493b5ff1fa04310

> Sounds like we've figured out something about the issue. Is it that weird username that my company laptop is using perhaps? Baffled.

its because of the corporate domain/AD setup, the `username()` function is not able to determine when to use the full username with the domain part, like you mentioned in a earlier comment
> Is func username() on line 157 in preflight_windows.go returning the username in a way that my computer is not expecting?

the `username()` function checks if the machine is part of a domain and returns a name like `GMX\6J1943897` if domain joined and just the username `6J1943897` if not part of domain, in your case even though the machine seems to be domain joined its returning the username without the domain part.

can you share the o/p of following:
```
PS> (Get-WmiObject -Class Win32_ComputerSystem).PartOfDomain
```

do you have a local administrator account on the machine?

edit: also share the o/p of `$env:ComputerName`

**Comment by brucedkyle** [comment]
@anjannath Thanks for your inquiry.

I do have admin access to the machine.

` (Get-WmiObject -Class Win32_ComputerSystem).PartOfDomain` returns
```
False
```
`echo $env:ComputerName` returns
```
LAPTOP-SLIV8CB4
```
`echo $env:UserDomain` returns
```
GMX
```

**Comment by anjannath** [comment]
@brucedkyle Thanks for trying it out and providing the feedback, i have another [test build](https://github.com/crc-org/crc/suites/11856368727/artifacts/619823114) for you to try, please cleanup/uninstall any existing installation of CRC

also confused as to why for you even though the machine is domain joined, the o/p for `(Get-WmiObject -Class Win32_ComputerSystem).PartOfDomain` is `False`

can you also provide the o/p of following commands after installing the MSI:
```
PS> Get-LocalGroupMember -Group 'crc-users'

PS> Get-LocalGroupMember -SID 'S-1-5-32-578'
```

**Comment by brucedkyle** [comment]
@anjannath Thanks so much for the update. We're a LOT closer.

I ran `crc.exe setup` and got:

```
INFO Using bundle path C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle exists
INFO Getting bundle for the CRC executable
INFO Downloading bundle: C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle...
2.98 GiB / 2.98 GiB [----------------------------------------------------------------------------------------] 100.00% 10.15 MiB p/s
INFO Uncompressing C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
crc.vhdx:  13.21 GiB / 13.21 GiB [-----------------------------------------------------------------------------------------] 100.00%
oc.exe:  99.99 MiB / 99.99 MiB [-------------------------------------------------------------------------------------------] 100.00%
INFO Checking if the daemon task is installed
INFO Installing the daemon task
INFO Checking if the daemon task is running
INFO Running the daemon task
INFO Checking admin helper service is running
Your system is correctly setup for using CRC. Use 'crc start' to start the instance
```

Next, I ran `crc.exe start` and got:

```
INFO Checking if running in a shell with administrator rights
INFO Checking Windows release
INFO Checking Windows edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if Hyper-V service is enabled
INFO Checking if crc-users group exists
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Checking if vsock is correctly configured
INFO Checking if the daemon task is installed
INFO Checking if the daemon task is running
WARN Preflight checks failed during `crc start`, please try to run `crc setup` first in case you haven't done so yet
expected crcDaemon task to be in 'Running' but got 'Ready
'
```

When I run ` Get-LocalGroupMember -Group 'crc-users'` I get:

```
ObjectClass Name          PrincipalSource
----------- ----          ---------------
User        GMX\6J1943897 AzureAD
```

When I run ` Get-LocalGroupMember -SID 'S-1-5-32-578'` I get:

```
ObjectClass Name          PrincipalSource
----------- ----          ---------------
User        GMX\6J1943897 AzureAD
```

I am guessing, but do no know for sure that the wonky set up (of not being assigned to a domain, while having a domain name like this) could be a security feature. But I don't really know for sure.

In any case, we are really close. Crazy that the INFO reports `is running` but the Preflight check reports `'Ready'`.

I appreciate your help.

**Comment by anjannath** [comment]
@brucedkyle thanks again for trying it out and providing the information, `crc cleanup` and then `crc setup` should fix the issue with the crcDaemon task.

about the user getting added to the right group is also working the issue was that crc setup was not detecting it correctly, which is also working now, we'll have the fix in for 2.17.0 release, thanks!

**Comment by brucedkyle** [comment]
@anjannath Unfortunately I am getting the same warning on startup `Preflight checks failed during 'crc start'`. But it appears crc machine does not exist.

```
PS C:\> crc cleanup
INFO Removing the daemon task
INFO Removing hosts file records added by CRC
INFO Removing pull secret from the keyring
INFO Removing older logs
INFO Removing CRC Machine Instance directory
INFO Removing crc's virtual machine
Cleanup finished
PS C:\> crc setup
INFO Using bundle path C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Checking if CRC bundle is extracted in '$HOME/.crc'
INFO Checking if C:\Users\6J1943897\.crc\cache\crc_hyperv_4.12.5_amd64.crcbundle exists
INFO Checking if the daemon task is installed
INFO Installing the daemon task
INFO Checking if the daemon task is running
INFO Running the daemon task
INFO Checking admin helper service is running
Your system is correctly setup for using CRC. Use 'crc start' to start the instance
PS C:\> crc start
INFO Checking if running in a shell with administrator rights
INFO Checking Windows release
INFO Checking Windows edition
INFO Checking if Hyper-V is installed and operational
INFO Checking if Hyper-V service is enabled
INFO Checking if crc-users group exists
INFO Checking if current user is in crc-users and Hyper-V admins group
INFO Checking if vsock is correctly configured
INFO Checking if the daemon task is installed
INFO Checking if the daemon task is running
WARN Preflight checks failed during `crc start`, please try to run `crc setup` first in case you haven't done so yet
expected crcDaemon task to be in 'Running' but got 'Ready
'
PS C:\>
```

I ran `crc status` and got:

```
Machine does not exist. Use 'crc start' to create it
```

What should I be doing?

**Comment by anjannath** [comment]
@brucedkyle from the logs its not clear why the `crcDaemon` task is not running, can you provide the o/p for `crc cleanup --log-level debug` and `crc setup --log-level debug`

you can also manually remove the crcDaemon task from the task scheduler if `crc cleanup` is not able to remove it, if you still face issue with the `crcDaemon` task please file a new issue for it, this is unrelated to the domain-user bug you had previously.

**Comment by brucedkyle** [comment]
@anjannath Thanks for your help with the domain-user issue. Much appreciated.

The crcDaemon task does not seem to exist. I'll close this one and raise a separate issue.

**Comment by andikaopp** [comment]
when i run `crc setup` i got this error the setup, could you to help me solve this issue?
![image](https://github.com/crc-org/crc/assets/18736601/51b611ef-7975-4161-9d60-f9147eee72f3)

**Comment by anjannath** [comment]
@andikaopp As the messages says, after installing the MSI you have to reboot for installation to finish and then run `crc setup`.

Since we'll need more information to debug, Please file a new issue about this and fill up the relevant information in the issue template.
